tests/testthat/test-paste_linter.R

```diff
@@ -249,6 +249,16 @@ test_that("paste0(collapse=...) is caught", {
   expect_lint("paste0(foo(x, y, z), collapse = '')", lint_msg, linter)
 })
 
+test_that("paste0(collapse=...) is caught in pipes", {
+  linter <- paste_linter()
+  lint_msg <- rex::rex("Use paste(), not paste0(), to collapse a character vector when sep= is not used.")
+
+  expect_lint('x %>% paste0(y, collapse = "")', NULL, linter)
```

Comment:
There is a discrepancy here between pipes for code that produces identical results:

``` r
library(magrittr)
library(lintr)

x <- letters[1:2]
x <- x %>% paste0(x, collapse = "")
lint(text = 'x <- x %>% paste0(x, collapse = "")', linters = paste_linter())


x <- letters[1:2]
x %<>% paste0(x, collapse = "")
lint(text = 'x %<>% paste0(collapse = "")', linters = paste_linter())
#> <text>:1:8: warning: [paste_linter] Use paste(), not paste0(), to collapse a character vector when sep= is not used.
#> x %<>% paste0(collapse = "")
#>        ^~~~~~~~~~~~~~~~~~~~~
```

<sup>Created on 2024-05-06 with [reprex v2.1.0](https://reprex.tidyverse.org)</sup>

Reply:
Maybe just a typo? The code in the latter example doesn't match the linted code.

Reply:
Ah, so dumb 🤦 

Anyway, should we include test for other pipes? I don't think it's needed, but just wanted to check.

``` r
library(lintr)
lint(text = 'x %<>% paste0(collapse = "")', linters = paste_linter())
#> <text>:1:8: warning: [paste_linter] Use paste(), not paste0(), to collapse a character vector when sep= is not used.
#> x %<>% paste0(collapse = "")
#>        ^~~~~~~~~~~~~~~~~~~~~
```

<sup>Created on 2024-05-06 with [reprex v2.1.0](https://reprex.tidyverse.org)</sup>

Reply:
OK, SG; done.